src/riskmatrix/views/risk_assessment.py

```diff
@@ -296,112 +321,115 @@ def __html__(self) -> str:
+    for color, indices in colors.items():
+        for index in indices:
+            i, j = divmod(index, 5)
+
+            fig.add_shape(
+                type="rect",
+                x0=j,
+                y0=4 - i,
+                x1=j + 1,
+                y1=5 - i,
+                line={"color": "white", "width": 0},
+                fillcolor=color,
+                layer="below",
             )
 
+    # Plot points
+    for risk in list(risks):
+        if risk.likelihood and risk.impact:
+            x, y = float(risk.likelihood - 1), float(4 - (risk.impact - 1))
+
+            # Adjust the position within the cell to avoid overlap
+            noise_x = np.random.uniform(0.1, 0.9)
+            noise_y = np.random.uniform(0.1, 0.9)
+            x, y = x + noise_x, y + noise_y
+
+            fig.add_trace(
+                go.Scatter(
+                    x=[x],
+                    y=[y],
+                    text=[str(risk.nr)],
+                    name="",
+                    mode="markers+text",
+                    marker={"color": "black", "size": 18},
+                    textposition="middle center",
+                    hoverinfo="text",
+                    hovertemplate=f"{risk.nr} {risk.name} \
+                    (Impact: {risk.impact} Likelihood: {risk.likelihood})",
+                    textfont={"color": "white"},
+                )
+            )
+
+    fig.update_xaxes(fixedrange=True)
+    fig.update_yaxes(fixedrange=True)
+
+    # Update layout
+    fig.update_layout(
+        xaxis={
+            "showgrid": False,
+            "zeroline": False,
+            "showticklabels": False,
+            "range": [-0.5, 5]
+        },
+        yaxis={
+            "showgrid": False,
+            "zeroline": False,
+            "showticklabels": False,
+            "range": [-0.5, 5]
+        },
+        showlegend=False,
+        width=700,
+        height=700,
+        margin={"l": 5, "r": 5, "t": 5, "b": 5},  # Adjusted margins
+        paper_bgcolor="white",
+        plot_bgcolor="white",
+    )
+
+    # Add axis labels
+    fig.add_annotation(
+        x=2.5, y=-0.2, text="Impact", showarrow=False, font={"size": 20}
+    )
+    fig.add_annotation(
+        x=-0.2,
+        y=2.5,
+        text="Likelihood",
+        showarrow=False,
+        textangle=-90,
+        font={"size": 20},
+    )
+
+    return fig.to_html(
+        full_html=False,
+        include_plotlyjs=True,
+        config={
+            "modeBarButtonsToRemove": ["zoom", "pan", "select", "lasso2d"]
+        },
+    )
+
+
+def generate_risk_matrix_view(
+    context: "Organization", request: "IRequest"
+) -> "RenderData":
+    table = AssessmentOverviewTable(context, request)
+
     return {
-        'title': _('Risk Matrix'),
-        'cells': cells,
-        'assessments': assessments,
+        "title": _("Risk Matrix"),
+        "plot": Markup(plot_risk_matrix(table.query())),  # noqa: MS001
+        "table": table,
```

Comment:
```suggestion
        'title': _('Risk Matrix'),
        'plot': plot_risk_matrix(table.query()),
        'table': table,
```
style nit, also I would return `Markup` from `plot_risk_matrix`, don't wrap it here.